### PR TITLE
Autopause clusters with busy kernels after 24 hours

### PR DESCRIPTION
**Pull request by akarukappadath**
\<your comments for this PR go here\>

Have you read [CONTRIBUTING.md](https://github.com/DataBiosphere/leonardo/blob/develop/CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've documented my API changes in Swagger

In all cases:

- [ ] Get a thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green
- [ ] Squash and merge; you can delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

**Comment by akarukappadath** [comment]
jenkins retest

**Comment by akarukappadath** [comment]
jenkins retest

**Comment by akarukappadath** [comment]
jenkins retest

**Comment by akarukappadath** [comment]
jenkins retest

**Comment by akarukappadath** [comment]
jenkins retest

File: src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/ClusterDateAccessedActor.scala [hunk 1]
```diff
@@ -48,6 +48,7 @@ class ClusterDateAccessedActor(autoFreezeConfig: AutoFreezeConfig, dbRef: DbRefe
   private def updateDateAccessed(clusterName: ClusterName, googleProject: GoogleProject, dateAccessed: Instant) = {
     logger.info(s"Update dateAccessed for $clusterName in project $googleProject to $dateAccessed")
     dbRef.inTransaction { dataAccess =>
+      dataAccess.clusterQuery.clearKernelFoundBusyDateByProjectAndName(googleProject, clusterName)
```
Comment (Qi77Qi) [inline src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/ClusterDateAccessedActor.scala]: I'm not familiar with slick, will this line actually get run? cc @rtitle  (this DBIO is not chained in anything)

**Review by Qi77Qi — COMMENTED** [review]
Would be nice to have https://broadworkbench.atlassian.net/browse/IA-983 rolled out before we make this change live

File: src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/ClusterComponent.scala [hunk 2]
```diff
@@ -336,6 +338,21 @@ trait ClusterComponent extends LeoComponent {
       findByIdQuery(id).filter { _.dateAccessed < Timestamp.from(dateAccessed)}.map(_.dateAccessed).update(Timestamp.from(dateAccessed))
     }
 
+    def clearKernelFoundBusyDate(id: Long) = {
```
Comment (Qi77Qi) [inline src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/ClusterComponent.scala]: can you add return type

File: src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/ClusterComponent.scala [hunk 3]
```diff
@@ -336,6 +338,21 @@ trait ClusterComponent extends LeoComponent {
...
+      findByIdQuery(id).map(_.kernelFoundBusyDate).update(None)
+    }
+
+    def updateKernelFoundBusyDate(id: Long, kernelFoundBusyDate: Instant): DBIO[Int] = {
```
Comment (Qi77Qi) [inline src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/ClusterComponent.scala]: what does returned `Int` mean?

File: src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/ClusterComponent.scala [hunk 4]
```diff
@@ -336,6 +338,21 @@ trait ClusterComponent extends LeoComponent {
...
+      findByIdQuery(id).map(_.kernelFoundBusyDate).update(Option(Timestamp.from(kernelFoundBusyDate)))
+    }
+
+    def clearKernelFoundBusyDateByProjectAndName(googleProject: GoogleProject, clusterName: ClusterName): DBIO[Int] = {
```
Comment (Qi77Qi) [inline src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/ClusterComponent.scala]: what does returned Int mean? A comment would be nice

**Review by Qi77Qi — APPROVED** [review]
left a minor comments, otherwise 👍

**Pull request closed without merging**